ArduPlane/navigation.cpp

```diff
@@ -127,15 +127,15 @@ void Plane::calc_airspeed_errors()
     // Set target to current airspeed + ground speed undershoot,
     // but only when this is faster than the target airspeed commanded
     // above.
-    if (control_mode >= FLY_BY_WIRE_B && (aparm.min_gndspeed_cm > 0)) {
+    if (auto_throttle_mode && (aparm.min_gndspeed_cm > 0)) {
```

Comment:
I think it's correct but this is a behavior change for CIRCLE.

Reply:
as noted in the description. Although, CIRCLE would be pretty terrible if it wasn't making forward progress.. not much of a circle. Also, CIRCLE is mainly used in failsafe modes. 

This actually looks like I'm uncovering a bug. groundspeed_undershoot gets set only when there's a GPS.log >= 2D but never resets on GPS loss. The undershoot sticks forever. Same goes with if you change min_gndspeed_cm to zero during flight but at least its only used later when min_gndspeed_cm is >0.

Reply:
Circle doesn't actually draw a circle, it's actually a constant bank mode and doesn't care about drawing a circle on the ground. Given this thought I wonder if it really should just ignore the groundspeed. It's not actually maintaining a real circle, and is basically just a failsafe mode that shouldn't fall out of the sky but can wander wherever I think it's better that this maintain the airspeed rather then care about ground speed.

Reply:
yup, fair enough. I'll put an exception in there for CIRCLE.